src/app/components/task-button/task-button.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { ConstantsService } from 'src/app/services/constants.service';
+import { TaskService } from 'src/app/services/task.service';
+
+@Component({
+  selector: 'app-task-button',
+  templateUrl: './task-button.component.html',
+  styleUrls: ['./task-button.component.css'],
+})
+export class TaskButtonComponent implements OnInit {
+  @Input() id!: number;
```

Comment:
Why do we expect id to be `number|undefined` ?

Reply:
updated with initial value